lib/iris/__init__.py

```diff
@@ -427,4 +428,10 @@ def load_raw(uris, constraints=None, callback=None):
 
 def sample_data_path(*path_to_join):
     """Given the sample data resource, returns the full path to the file."""
-    return os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    target = os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    if not glob.glob(target):
```

Comment:
The whole point of the function is to be able to supply a relative path and not have to care about the absolute path to the samples directory, so absolute paths have no place in the arguments to this function. If someone is doing `sample_data_path('/full/path/to/samples/air_temp.pp')` then they are misusing this function. The new exception should discourage them from such behaviour!